src/Core/FeatureFlag/FeatureFlagManager.php

```diff
@@ -34,31 +34,29 @@
 use PrestaShop\PrestaShop\Core\FeatureFlag\Layer\QueryLayer;
 use PrestaShopBundle\Entity\Repository\FeatureFlagRepository;
 use Psr\Container\ContainerInterface;
-use Symfony\Contracts\Service\ServiceSubscriberInterface;
+use Symfony\Component\DependencyInjection\Attribute\TaggedLocator;
 
-class FeatureFlagManager implements ServiceSubscriberInterface, FeatureFlagStateCheckerInterface
+class FeatureFlagManager implements FeatureFlagStateCheckerInterface
 {
+    /**
+     * List of layers used by the feature flag.
+     *
+     * @var array<string, string>
+     */
+    private const LAYERS = [
+        FeatureFlagSettings::TYPE_ENV => EnvLayer::class,
+        FeatureFlagSettings::TYPE_DOTENV => DotEnvLayer::class,
+        FeatureFlagSettings::TYPE_DB => DbLayer::class,
+        FeatureFlagSettings::TYPE_QUERY => QueryLayer::class,
+    ];
+
     public function __construct(
+        #[TaggedLocator('ps.feature_flag_layer')]
```

Comment:
Did not know about it, but very useful
https://jolicode.com/blog/how-taggedlocator-can-help-you-design-better-symfony-application